src/sentry/api/event_search.py

```diff
@@ -387,6 +387,20 @@ class SearchConfig:
     # Enables boolean filtering (AND / OR)
     allow_boolean = True
 
+    # Allows us to specify an allowlist of keys we will accept for this search.
+    # If empty, allow all keys.
+    allowed_keys: Set[str] = field(default_factory=set)
+
+    # Which key we should return any free text under
+    free_text_key = "message"
+
+    @classmethod
+    def create_from(cls, search_config: "SearchConfig", **overrides):
```

Comment:
I kinda wish this part was a sep PR so I could approve it, and then defer to someone else to approve the queryset stuff in organization_releases.py

Reply:
I can split off the `organization_releases.py` stuff if you want. It should be identical, and it's covered by tests

Reply:
I'll approve, it looks fine, I'm just being pedantic.